scylla.go

```diff
@@ -4,17 +4,96 @@ import (
 	"fmt"
 	"math"
 	"strconv"
+	"strings"
 	"sync/atomic"
 )
 
 // scyllaSupported represents Scylla connection options as sent in SUPPORTED
 // frame.
+// FIXME: Should also follow `cqlProtocolExtension` interface.
 type scyllaSupported struct {
 	shard             int
 	nrShards          int
 	msbIgnore         uint64
 	partitioner       string
 	shardingAlgorithm string
+	lwtFlagMask       int
+}
+
+// CQL Protocol extension interface for Scylla.
+// Each extension is identified by a name and defines a way to serialize itself
+// in STARTUP message payload.
+type cqlProtocolExtension interface {
+	name() string
+	serialize() map[string]string
+}
+
+func findCQLProtoExtByName(exts []cqlProtocolExtension, name string) *cqlProtocolExtension {
```

Comment:
It's not very common to return a pointer to an interface in Go. That's the rationale for doing so here?

Reply:
What is the canonical way to lookup an object in a slice or map in Go? I can either return a pointer to the object or return an index, but in that case I would need to look it up again by doing `exts[i]` (probably doesn't incur any noticeable runtime cost, but anyway seems extraneous).

Reply:
The returned pointer implements the interface right? Then you can declare the function to return the interface but the code returns the actual pointer to the concrete type.

Reply:
Thanks for the suggestion. I didn't realize that interfaces can have a somewhat mixed semantics, either a pointer or a value, so that you can assign a `nil` to a variable of interface type. I will adjust.